lib/facter/consul_version.rb

```diff
@@ -0,0 +1,15 @@
+# consul_version.rb
+
+Facter.add(:consul_version) do
+  setcode do
+    if Facter.value(:kernel) == 'Linux'
+      version = Facter::Util::Resolution.exec('consul --version')
+      version = version.lines.first.split[1].tr('v','')
+    elsif Facter.value(:kernel) == 'windows'
+      version = Facter::Util::Resolution.exec('"C:\\Program Files\\Consul\\consul.exe" --version')
```

Comment:
Don't think it would be possible to parameterize it, as facts run on the agent prior to catalogue compilation, so don't think they have access to any of the puppet parameters.  Would probably have to ensure consul was on the Windows path once support is added.

Also, I wrote the windows puppet module for my companies use of Consul on Windows, depends on NSSM.  Happy to contribute to the work you're doing if you're willing to put it in an open repo somewhere?